library/Network/CloudSeeder/Interfaces.hs

```diff
@@ -238,65 +212,60 @@ computeChangeset' (StackName stackName) provisionType templateBody params tags =
     provisionTypeToChangeSetType CreateStack = CF.Create
     provisionTypeToChangeSetType (UpdateStack _) = CF.Update
 
-getStackInfo' :: MonadCloudIO r e m => StackName -> m (Maybe Stack)
-getStackInfo' (StackName stackName) = do
+describeStack' :: MonadCloudIO r e m => StackName -> m (Maybe Stack)
+describeStack' (StackName stackName) = do
   env <- ask
   let request = CF.describeStacks & CF.dStackName ?~ stackName
   runResourceT . runAWST env $ do
     response <- IO.trying_ (_StackDoesNotExistError (StackName stackName)) $ send request
     case response ^? _Just.CF.dsrsStacks of
       Nothing -> return Nothing
       Just [s] -> do
+        let awsOutputs = s ^. CF.sOutputs
+        outputs' <- M.fromList <$> mapM outputToTuple awsOutputs
         let awsParams = s ^. CF.sParameters
         params <- S.fromList <$> mapM awsParameterKey awsParams
-        let stack = Stack params
-        return $ Just stack
+        pure $ Just $ Stack
+          (s ^. CF.sStackStatusReason)
+          (s ^. CF.sChangeSetId)
+          (s ^. CF.sStackName)
+          outputs'
+          params
+          (s ^. CF.sStackId)
+          (s ^. CF.sStackStatus)
       Just _ -> throwing _CloudErrorInternal "describeStacks returned more than one stack"
   where
     awsParameterKey x = case x ^. CF.pParameterKey of
       (Just k) -> return k
       Nothing -> throwing _CloudErrorInternal "stack parameter key was missing"
-
-getStackOutputs' :: MonadCloudIO r e m => StackName -> m (Maybe (M.Map T.Text T.Text))
-getStackOutputs' (StackName stackName) = do
-    env <- ask
-    let request = CF.describeStacks & CF.dStackName ?~ stackName
-    runResourceT . runAWST env $ do
-      response <- IO.trying_ (_StackDoesNotExistError (StackName stackName)) $ send request
-      case response ^? _Just.CF.dsrsStacks of
-        Nothing -> return Nothing
-        Just [stack] -> Just . M.fromList <$> mapM outputToTuple (stack ^. CF.sOutputs)
-        Just _ -> throwing _CloudErrorInternal "describeStacks returned more than one stack"
-  where
     outputToTuple x = case (x ^. CF.oOutputKey, x ^. CF.oOutputValue) of
       (Just k, Just v) -> return (k, v)
       (Nothing, _) -> throwing _CloudErrorInternal "stack output key was missing"
       (_, Nothing) -> throwing _CloudErrorInternal "stack output value was missing"
 
-runChangeSet' :: MonadCloudIO r e m => T.Text -> m ()
+runChangeSet' :: MonadCloudIO r e m => T.Text -> m Int
```

Comment:
Is there a reason this returns a status code? It seems like it always returns `200` or throws, so it would make sense to keep returning `()`.